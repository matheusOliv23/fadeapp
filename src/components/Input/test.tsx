import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Input } from '.'

const { getByText, getByRole } = screen

describe('<Input />', () => {
  it('Should render Input', () => {
    render(<Input label="Nome" name="name" />)

    const input = getByRole('textbox', { name: 'Nome' })
    const label = getByText('Nome')

    expect(label).toBeInTheDocument()

    expect(input).toHaveAttribute('name', 'name')
    expect(input).toHaveAttribute('id', 'name')
  })

  it('Should change label styles', () => {
    const initialValue = ''
    render(<Input label="Nome" name="name" value={initialValue} />)

    const input = getByRole('textbox', { name: 'Nome' })
    const label = getByText('Nome')

    expect(label).toHaveClass('text-[1.125rem] leading-[1.3125rem] top-1')

    input.focus()

    expect(label).toHaveClass(
      'text-[0.75rem] top-[-0.215rem] leading-[0.875rem]',
    )

    input.blur()

    expect(label).toHaveClass('text-[1.125rem] leading-[1.3125rem] top-1')
  })

  it('Should change value on type', async () => {
    render(<Input label="Nome" name="name" value="" />)

    const text = 'John Doe'
    const input = getByRole('textbox', { name: 'Nome' })

    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveDisplayValue('John Doe')

      input.blur()
    })
  })

  it('Should match snapshot', () => {
    const { container } = render(<Input label="Nome" name="name" value="" />)

    expect(container).toMatchSnapshot()
  })
})
