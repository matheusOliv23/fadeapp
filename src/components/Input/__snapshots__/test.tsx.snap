// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should match snapshot 1`] = `
<div>
  <div
    class="
  box-border
  overflow-hidden
  max-h-max
  odd:rounded-t-[0.625rem]
  even:rounded-b-[0.625rem]
  even:border-t-0
  relative
  flex
  border
  bg-white
  border-solid 
  border-secondary 
  h-[2.875rem]
"
  >
    <input
      autocomplete="off"
      class="
  box-border
  outline-0
  overflow-hidden
  w-full
  absolute
  rounded
  bg-transparent
  transition-all
  duration-200
  ease-in-out
  pt-[16px]
  pl-[0.5rem]
  text-[1rem]
  leading-[1.1875rem]
  text-secondary
  bottom-[0.4rem]
"
      id="name"
      name="name"
      value=""
    />
    <label
      class="
            
  box-border
  overflow-hidden
  absolute
  text-gray-700
  p-2
  pointer-events-none 
  transition-all 
  duration-200 
  ease-in-out\\"

            top-1 text-[1.125rem] leading-[1.3125rem]
          "
      for="name"
    >
      Nome
    </label>
  </div>
</div>
`;
